tools/cli/commands/changelog.js

```diff
@@ -734,72 +737,100 @@ async function promptChangelog( argv, needChangelog, types ) {
 		value,
 		name: `[${ value.padEnd( maxLength, ' ' ) }] ${ name }`,
 	} ) );
-	const commands = await inquirer.prompt( [
-		{
-			type: 'string',
-			name: 'changelogName',
-			message: 'Name your changelog file:',
-			default: gitBranch,
-			validate: input => {
-				const fileExists = doesFilenameExist( input, needChangelog );
-				if ( fileExists ) {
-					return 'Please choose another file name, or delete the file manually.';
-				}
-				return true;
-			},
-		},
-		{
-			type: 'list',
-			name: 'significance',
-			message: 'Significance of the change, in the style of semantic versioning.',
-			choices: [
-				{
-					value: 'patch',
-					name: '[patch] Backwards-compatible bug fixes.',
-				},
-				{
-					value: 'minor',
-					name: '[minor] Added (or deprecated) functionality in a backwards-compatible manner.',
-				},
-				{
-					value: 'major',
-					name: '[major] Broke backwards compatibility in some way.',
-				},
-			],
-		},
-		{
-			type: 'list',
-			name: 'type',
-			message: 'Type of change.',
-			choices: choices,
+	// Get the changelog name.
+	const { changelogName } = await prompt( {
+		type: 'input',
+		name: 'changelogName',
+		message: 'Name your changelog file:',
+		default: gitBranch,
+		validate: input => {
+			const fileExists = doesFilenameExist( input, needChangelog );
+			if ( fileExists ) {
+				return 'Filename exists already. Please choose another file name, or delete the file existing manually.';
+			}
+			return true;
 		},
-		{
-			type: 'string',
-			name: 'entry',
-			message: 'Changelog entry. May be left empty if this change is particularly insignificant.',
-			when: answers => answers.significance === 'patch',
+	} );
+
+	// Get the significance.
+	const { significance } = await prompt( {
+		type: 'autocomplete',
+		name: 'significance',
+		message: 'Significance of the change, in the style of semantic versioning.',
+		suggest: ( input, significanceChoices = [ 'patch', 'minor', 'major' ] ) => {
+			return Promise.resolve(
+				significanceChoices.filter( choice => choice.value.startsWith( input ) )
+			);
 		},
```

Comment:
I think you overthought this.
```suggestion
		suggest: ( input, choices ) => choices.filter( choice => choice.value.startsWith( input ) ),
```

Although I note the highlighting enquirer applies is wonky, it doesn't match what's actually being searched. Maybe we should just turn it off.
```suggestion
		suggest: ( input, choices ) => choices.filter( choice => choice.value.startsWith( input ) ),
		highlight: v => v,
```

Reply:
Ah yeah I did indeed.